src/pa02/chutes_simulation.py

```diff
@@ -0,0 +1,261 @@
+    def __init__(self, board, dropped_steps=1):
+        super().__init__(board)
+        self.dropped_steps = dropped_steps
+        self.ladder_checker = False
+
+    """Subclass of Player that takes less steps on the next move after
+    climbing a ladder
+
+    Parameters
+    ----------
+    board - Board() object
+        choose what kind of board this player shall play on
+    dropped_steps - int
+        number of steps to drop
+    """
+
+    def move(self):
+        """LazyPLayer first checks if it climbed a ladder last move, it then
+         executes the new move depending on the answer"""
+        if self.ladder_checker:
+            random_number = random.randint(1, 6)
+            self.pos += max(0, random_number - self.dropped_steps)
+            if self.board.position_adjustment(self.pos) > 0:
+                self.ladder_checker = True
+            else:
+                self.ladder_checker = False
+            self.pos += self.board.position_adjustment(self.pos)
+        else:
+            self.pos += random.randint(1, 6)
+            if self.board.position_adjustment(self.pos) > 0:
+                self.ladder_checker = True
+            else:
+                self.ladder_checker = False
+            self.pos += self.board.position_adjustment(self.pos)
+
+
+class Simulation:
+    def __init__(self, player_field=(Player, LazyPlayer, LazyPlayer),
+                 board=Board(), seed=1337, randomize_players=True):
+        self.player_field = list(player_field)
+        self.board = board
+        self.seed = seed
+        self.randomize_players = randomize_players
+        self.results = []
+        """Initialise the simulation
+        Parameters
+        ---------
+        player_field : list/tuple
+            Names of the Player classes to use
+        board : Board() object
+            The board to use in the simulation
+        seed : int
+            Random generator seed
+        randomize_players : Bool
+            If the player_field should be shuffled before each game
+        """
+
+    def single_game(self):
+        """Performs a single game and returns the winning class name and the
+        number of moves
+
+        Returns
+        --------
+        tuple
+            tuple with winning class and number of moves
+        """
+        if self.randomize_players:
+            random.shuffle(self.player_field)
+        player_objects = [i(self.board) for i in self.player_field]
+        moves = 0
+        while True:
+            for i in player_objects:
+                i.move()
+                if self.board.goal_reached(i.pos):
+                    moves += 1
+                    return moves, type(i).__name__
+            moves += 1
+
+    def run_simulation(self, number_of_games):
+        """Runs a number of simulations and adds results to a list
+        Parameters
+        ----------
+        number_of_games - int
+            number of games to simulate
+        """
+        random.seed(self.seed)

```

Comment:
This should go to the init method, or otherwise the seed will get reset in each simulation